pkg/controller/controllercmd/builder.go

```diff
@@ -240,7 +240,7 @@ func (b *ControllerBuilder) Run(ctx context.Context, config *unstructured.Unstru
 
 	leaderElection.Callbacks.OnStartedLeading = func(ctx context.Context) {
 		if err := b.startFunc(ctx, controllerContext); err != nil {
-			klog.Fatal(err)
+			klog.Error(err)
```

Comment:
the call back is asynchronous so if you error instead of an exit is it gonna be still running dead branch?